src/configure/clients/github/githubClient.ts

```diff
@@ -1,69 +1,150 @@
 import { UrlBasedRequestPrepareOptions } from 'ms-rest';
+import { stringCompareFunction } from "../../helper/commonHelper";
 import { GitHubProvider } from "../../helper/gitHubHelper";
 import { SodiumLibHelper } from "../../helper/sodium/SodiumLibHelper";
+import { GitHubOrganization, GitHubRepo } from '../../model/models';
 import { RestClient } from "../restClient";
+const uuid = require('uuid/v4');
 
 const UserAgent = "deploy-to-azure-vscode";
 
 export class GithubClient {
 
     private patToken: string;
     private url: string;
+    private listOrgPromise: Promise<void | GitHubOrganization[]>;
+
 
     constructor(patToken: string, remoteUrl: string) {
         this.patToken = patToken;
         this.url = remoteUrl;
+        this.listOrgPromise = this.listOrganizations(this.patToken);
     }
 
-    public async createOrUpdateGithubSecret(secretName: string, body: string): Promise<void> {
-        let secretKeyObject: GitHubSecretKey = await this._getGitHubSecretKey();
-        let sodiumObj = new SodiumLibHelper(secretKeyObject.key);
-        let encryptedBytes: Uint8Array = sodiumObj.encrypt(body);
-        let encryptedBytesAsString: string = SodiumLibHelper.convertUint8ArrayToString(encryptedBytes);
-        let encryptedEncodedText = SodiumLibHelper.encodeToBase64(encryptedBytesAsString);
-        await this._setGithubSecret(secretName, secretKeyObject.key_id, encryptedEncodedText);
+    public createGithubRepo(orgName: string, repoName: string): Promise<void | GitHubRepo> {
+        let restClient =new RestClient();
+        let Url = "https://api.github.com/orgs/" + orgName + "/repos";
+        try{
```

Comment:
do we need a try-catch here as the exception would anyways be handled by caller?

Reply:
The caller is not handling the exception, it is just deciding which case to execute (out of the 3 cases for naming GitHub repo name) depending upon the return type of this function. By doing so, the code is shortened as we don't need nested try-catch blocks.